app/services/container_dashboard_service.rb

```diff
@@ -79,13 +79,9 @@ def status
   end
 
   def providers
-    provider_classes_to_ui_types = {
-      "ManageIQ::Providers::Openshift::ContainerManager"           => :openshift,
-      "ManageIQ::Providers::OpenshiftEnterprise::ContainerManager" => :openshift,
-      "ManageIQ::Providers::Kubernetes::ContainerManager"          => :kubernetes,
-      "ManageIQ::Providers::Atomic::ContainerManager"              => :atomic,
-      "ManageIQ::Providers::AtomicEnterprise::ContainerManager"    => :atomic
-    }
+    provider_classes_to_ui_types = ManageIQ::Providers::ContainerManager.subclasses.each_with_object({}) { |subclass, h|
+      name = subclass.name.split('::')[2]
+      h[subclass.name] = name.underscore.downcase.to_sym }
```

Comment:
I think we really should have a method for these things. Either in the class itself or in some decorator. I can see similar code on more places and frankly, this is not good.